src/EditorFeatures/Test2/IntelliSense/IntellisenseQuickInfoBuilderTests.vb

```diff
@@ -0,0 +1,119 @@
+                                    New TaggedText("Method", "WriteLine"),
+                                    New TaggedText("Punctuation", "("),
+                                    New TaggedText("Keyword", "string"),
+                                    New TaggedText("Space", " "),
+                                    New TaggedText("Parameter", "value"),
+                                    New TaggedText("Punctuation", ")"),
+                                    New TaggedText("Space", " "),
+                                    New TaggedText("Punctuation", "("),
+                                    New TaggedText("Punctuation", "+"),
+                                    New TaggedText("Space", " "),
+                                    New TaggedText("Text", "18"),
+                                    New TaggedText("Space", " "),
+                                    New TaggedText("Text", "overloads"),
+                                    New TaggedText("Punctuation", ")")})),
+                            QuickInfoSection.Create("DocumentationComments",
+                                ImmutableArray.Create({New TaggedText("Text", "Writes the specified string value, followed by the current line terminator, to the standard output stream.")})),
+                            QuickInfoSection.Create("Exception",
+                                ImmutableArray.Create({
+                                    New TaggedText("Text", "Exceptions"),
+                                    New TaggedText("LineBreak", "\r\n"),
+                                    New TaggedText("Space", " "),
+                                    New TaggedText("Namespace", "System"),
+                                    New TaggedText("Punctuation", "."),
+                                    New TaggedText("Namespace", "IO"),
+                                    New TaggedText("Punctuation", "."),
+                                    New TaggedText("Class", "IOException")}))}))
+
+            Dim view = New Mock(Of ITextView) With {
+                .DefaultValue = DefaultValue.Mock
+            }
+
+            Dim textVersion = view.Object.TextSnapshot.Version
+            Dim trackingSpan = textVersion.CreateTrackingSpan(codeAnalysisQuickInfoItem.Span.ToSpan(), SpanTrackingMode.EdgeInclusive)
+
+            Dim intellisenseQuickInfo = IntellisenseQuickInfoBuilder.BuildItem(trackingSpan, codeAnalysisQuickInfoItem)
+
+            Assert.NotNull(intellisenseQuickInfo)
+
+            Assert.IsType(Of Adornments.ContainerElement)(intellisenseQuickInfo.Item)
+            Dim container = CType(intellisenseQuickInfo.Item, Adornments.ContainerElement)
+            Assert.Equal(3, container.Elements.Count())
+            Assert.Equal(ContainerElementStyle.Stacked, container.Style)
+
+
+            Assert.IsType(Of Adornments.ContainerElement)(container.Elements.ElementAt(0))
+            Dim firstRowContainer = CType(container.Elements.ElementAt(0), Adornments.ContainerElement)
```

Comment:
You might find Assert.Collection() handy here. It's an xunit thing that will grab each element, and then give you a function that it calls. So you can say something like.
```
Assert.Container(container.Elements,
    Sub(firstRowContainer),
        (your existing asserts
    End Sub)
```

etc. etc.

Reply:
We don't use it often (I think because people don't know about it), but if it makes things worse feel free to disregard this. It's sometimes cute, sometimes terrible.

Reply:
changed. It looks so much nicer now. Thanks :)